File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        left = 0
        right = len(s) - 1
    
        
        while(left < right):
            if not s[right].isalnum():
                right-=1
            elif not s[left].isalnum(): 
                left+= 1
            else:
                if(s[right].lower()) != (s[left].lower()): 
                    return False
                right-=1
                left+= 1
        return True
            
        

File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        int left = 0;
        int right = s.size() - 1;
        bool ans = true; 
        
        while(left < right){
            if(isalpha(s[right]) || isdigit(s[right])){
                if(isalpha(s[left]) || isdigit(s[left])){
                    if(tolower(s[right]) != tolower(s[left])) return !ans; 
                    else{
                        right--; 
                        left++;
                    }
                }
                else{
                    left++; 
                }
            }
            else{right--;} 
        }

        return ans; 
        
        
    }
};